pkg/virtctl/imageupload/imageupload.go

```diff
@@ -177,6 +179,22 @@ func parseArgs(args []string) error {
 		return fmt.Errorf("cannot upload to a readonly volume, use either ReadWriteOnce or ReadWriteMany if supported")
 	}
 
+	if volumeMode != "" {
+		if volumeMode != string(v1.PersistentVolumeBlock) && volumeMode != string(v1.PersistentVolumeFilesystem) {
+			return fmt.Errorf("%s is not supported for volumeMode, use either Block or Filesystem!", volumeMode)
+		}
+	}
+
+	if blockVolume && volumeMode == "" {
+		fmt.Printf("--block-volume is deprecated, use --volume-mode instead\n")
+		volumeMode = string(v1.PersistentVolumeBlock)
+	}
+
+	if !blockVolume && volumeMode == "" {
+		fmt.Printf("--block-volume is deprecated, use --volume-mode instead\n")
+		volumeMode = string(v1.PersistentVolumeFilesystem)
+	}
+
```

Comment:
Thanks. `parseArgs` is already 40 lines long, and does not fit my terminal. Please consider extracting these lines to a separate function.

Reply:
Done, move to `normalizeVolumeMode`.